### PR TITLE
Integration with treemacs based buffers

### PR DESCRIPTION
**Pull request by yyoncho**

### CODE REVIEW

File: link-hint.el [hunk 1]
```diff
@@ -73,7 +73,8 @@
     ;; generic
     link-hint-button
     link-hint-text-url
-    link-hint-file-link)
+    link-hint-file-link
+    link-hint-treemacs)
```
Comment (noctuid) [inline link-hint.el]: The generic types of links go at the bottom. Even though you've disabled the generic button type in treemacs mode, could you putton the new type before the "generic" comment (e.g. right after the woman button type) for consistency?

File: link-hint.el [hunk 2]
```diff
@@ -418,6 +419,26 @@ Only search the range between just after the point and BOUND."
   :open-multiple t
   :copy #'kill-new)
```
Comment (noctuid) [inline link-hint.el]: Please add a new heading comment before the new type functions and definition (`;; ** Treemacs Link`).

File: link-hint.el [hunk 3]
```diff
@@ -418,6 +419,26 @@ Only search the range between just after the point and BOUND."
...
 
+(defun link-hint--treemacs-open ()
```
Comment (noctuid) [inline link-hint.el]: The name of most of the helpers for opening is something allong the lines of `link-hint--open-<type>`. Could you call this something like `link-hint--open-treemacs-button` or `link-hint--open-treemacs-entry`?

File: link-hint.el [hunk 4]
```diff
@@ -418,6 +419,26 @@ Only search the range between just after the point and BOUND."
...
+  "Open in treemacs basesed buffer."
+  (if (string-prefix-p " *Treemacs-" (buffer-name))
+      (treemacs-visit-node-in-most-recently-used-window)
+    (treemacs-visit-node-default)))
+
+(link-hint-define-type 'treemacs
+  :next #'link-hint--next-button
+  :at-point-p (lambda ()
```
Comment (noctuid) [inline link-hint.el]: I'd prefer if you defined functions for everything.

File: link-hint.el [hunk 5]
```diff
@@ -418,6 +419,26 @@ Only search the range between just after the point and BOUND."
   :open-multiple t
   :copy #'kill-new)
 
+(defun link-hint--treemacs-open ()
+  "Open in treemacs basesed buffer."
```
Comment (noctuid) [inline link-hint.el]: Minor typo. Maybe change docstring to something like "Open an entry in a treemacs buffer" or "Open a file or folder in a treemacs buffer"

File: link-hint.el [hunk 6]
```diff
@@ -418,6 +419,26 @@ Only search the range between just after the point and BOUND."
...
+  (if (string-prefix-p " *Treemacs-" (buffer-name))
+      (treemacs-visit-node-in-most-recently-used-window)
+    (treemacs-visit-node-default)))
```
Comment (noctuid) [inline link-hint.el]: When would this condition be false?

Comment (yyoncho) [reply link-hint.el]: This is for https://github.com/emacs-lsp/lsp-treemacs . Also, dap-mode will follow with replacing tree-widget with treemacs.

**Comment by yyoncho** [comment]
Pushed a new version.

**Comment by noctuid** [comment]
Thanks!

**Pull request merged**